src/Redux/Actions/PostAction.js

```diff
@@ -35,32 +33,25 @@ export const getAllPosts = (token) => {
           },
         },
       );
+      let result = null;
       if (response.status === 200) {
-        let result = await response.json();
-        await dispatch({
-          type: SUCCEED_TO_GET_ALLPOST,
-          payload: result.posts,
-        });
-        return result.posts;
+        result = await response.json();
+        await dispatch(getAllPostsSuccess(result.posts));
       } else {
-        await dispatch({
-          type: FAILED_TO_GET_ALLPOST,
-          payload: null,
-        });
+        await dispatch(getAllPostsFail());
       }
+      return result ? result.posts : null;
```

Comment:
앞에서 result null로 선언했자네유?
```suggestion
      return result.posts;
```

Reply:
네......